manifests/bundle.pp

```diff
@@ -87,7 +87,7 @@
   }
 
   if $multicore {
-    if $multicore == '0' or $multicore >= $::processorcount {
+    if $multicore == '0' or versioncmp($multicore,$::processorcount) >= 0 {
```

Comment:
To protect against multicore being specified as an integer, you could cast it as a string.

```puppet
if "${multicore}" == '0' or versioncmp("${multicore}",$::processorcount) >= 0 {
```